File: Java/T12/src/Lampada.java

public class Lampada {
	boolean estado=false;
	int limiteclicks;
	
	void clicklampada() {
		System.out.println("Click");
		if(limiteclicks > 0) {
			if(estado) {
				estado = false;
				limiteclicks--;
			}			
			else 
			{
				estado = true;
				limiteclicks--;
			}
		}
		else {
			System.out.println("Lampada Queimou!!");
			estado = false;
		}			
	}
	
	void estadoLampada(){
		if(estado)
			System.out.println("Lampada Ligada");
		else
			System.out.println("lampada Desligada");
	}
	
	
	
}

File: Scala/T2/Ex2/src/Ex2/Lampada.scala
package Ex2

class Lampada {
  var estado = false
  var limiteclicks = 0

  def clicklampada(): Unit = {
    println("Click")
    if (limiteclicks > 0) if (estado) {
      estado = false
      limiteclicks -= 1
    }
    else {
      estado = true
      limiteclicks -= 1
    }
    else {
      println("Lampada Queimou!!")
      estado = false
    }
  }

  def estadoLampada(): Unit = {
    if (estado) println("Lampada Ligada")
    else println("Lampada Desligada")
  }
}